x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java

```diff
@@ -488,5 +462,31 @@ static void writeSearchApplicationBinaryWithVersion(SearchApplication app, Outpu
         }
     }
 
+    static class DelegatingIndexNotFoundActionListener<T, R> extends ActionListener.Delegating<T, R> {
```

Comment:
Note that `ActionListener.Delegating` is now `org.elasticsearch.action.DelegatingActionListener` in `main` (#94320) so you'll get some conflict or compile error here after merging this and `main`.